pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```
@@ -3180,12 +3180,56 @@ public TableStats getTableStats(String tableNameWithType) {
+   *    would be returned. Intersection is taken since the method guarantees to return
+   *    brokers which can serve the given table. In case of no type provided, it returns
+   *    broker which can serve both table-types.
+   *
+   * @param tableName name of table with or without type suffix.
+   * @return list of brokers serving the given table in the format: Broker_hostname_port.
+   * @throws TableNotFoundException when no table exists with the given name.
+   */
+  public List<String> getLiveBrokersForTable(String tableName)
+      throws TableNotFoundException {
+    TableType inputTableType = TableNameBuilder.getTableTypeFromTableName(tableName);
+    if (inputTableType != null) {
+      if (!hasTable(tableName)) {
+        throw new TableNotFoundException(String.format("Table=%s not found", tableName));
+      }
+      return getLiveBrokersForTable(tableName, inputTableType);
+    }
+    boolean hasOfflineTable = hasOfflineTable(tableName);
+    boolean hasRealtimeTable = hasRealtimeTable(tableName);
+    if (!hasOfflineTable && !hasRealtimeTable) {
+      throw new TableNotFoundException(String.format("Table=%s not found", tableName));
+    }
+    if (hasOfflineTable && hasRealtimeTable) {
+      Set<String> offlineTables = new HashSet<>(getLiveBrokersForTable(tableName, TableType.OFFLINE));
+      return getLiveBrokersForTable(tableName, TableType.REALTIME).stream()
+          .filter(offlineTables::contains)
+          .collect(Collectors.toList());
+    } else {
+      return getLiveBrokersForTable(tableName, hasOfflineTable ? TableType.OFFLINE : TableType.REALTIME);
+    }
+  }
+
   /**
    * Return the list of live brokers serving the corresponding table.
    *  Each entry in the broker list is of the following format:
    *  Broker_hostname_port
    */
-  public List<String> getLiveBrokersForTable(String tableNameWithType) {
+  public List<String> getLiveBrokersForTable(String tableName, TableType tableType) {
```

Suggest merging the logic of this method into the main method to avoid looking up `BROKER_RESOURCE` twice. Also (not introduced in this PR), when `BROKER_RESOURCE` cannot be found, we should probably throw an exception instead of returning empty result